### DISCUSSION TITLE
Default syntax-block type for own syntax

### DISCUSSION THREAD

**Discussion by jmeier** [idaholab/moose#27960] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Dear Moose Community,

For an custom syntax-block I'd like to define a "default type" so the user can skip to enter "type = ...". Please note the current state of my input file syntax:

```
[MyBlock]
  type = 'MyBlock'   # <-- how can I ask Moose to try to use the block name as type? Like e.g. Moose is doing the for "Functions"

  [SubBlock0]
    type = 'MySubBlock'  # <-- can I define a default subblock-type for MyBlock so I can skip this line?
    some_parameter = 0.0
  []

  [SubBlock1]
    type = 'MySubBlock'  # <-- can I define a default subblock-type for MyBlock so I can skip this line?
    some_parameter = 0.4
  []

[]
```

To define "MyBlock" an "MySubBlock" I used registerMooseObjectTask and registerSyntaxTask.

Thank you!

**Comment by GiudGiud** [comment]
Hello

If you only have a single object type you care about in a given syntax then in the Action (likely the AddMyBlockAction) you can not use a type and always create the same object

Guillaume

**Reply by jmeier** [reply]
Hello

If I omit the type parameter, Moose gives "missing required parameter 'MyBlock/MySubBlock/type'". This despite the fact, that I ignore the type parameter in AddMyBlockAction and AddMySubBlockAction.

Jörg

**Reply by GiudGiud** [reply, marked as answer]
Is your action derived from MooseObjectAction? If so the 'type' is required. You can change that in validParams

**Reply by jmeier** [reply]
Dear Guillaume,

Wonderful! That was it. Thank you very much!

Jörg
